CMakeLists.txt

```diff
@@ -825,7 +826,9 @@ IF(XNNPACK_BUILD_LIBRARY)
   TARGET_INCLUDE_DIRECTORIES(operator-utils PRIVATE include src)
   TARGET_INCLUDE_DIRECTORIES(memory PRIVATE include src)
   TARGET_INCLUDE_DIRECTORIES(mutex PRIVATE include src)
-  TARGET_INCLUDE_DIRECTORIES(post-operation PUBLIC include src)
+  TARGET_INCLUDE_DIRECTORIES(post-operation PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
```

Comment:
Why do you need to specify `$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>` instead of just `include`? Why do you do it only for some targets (`XNNPACK`, and `post-operation`)?

Reply:
If we use

```cmake
TARGET_INCLUDE_DIRECTORIES(XNNPACK PUBLIC include)
```

with the CMake configuration which allows discovery of XNNPACK after installation, then trying to build will give the following error:

```console
CMake Error in CMakeLists.txt:
  Target "XNNPACK" INTERFACE_INCLUDE_DIRECTORIES property contains path:
     "/build/source/include"
  which is prefixed in the source directory.
```

CMake interprets `include` as a relative path and converts it to an absolute path based on the current working directory. This path would double as the install path, which is not what we want (that build path wouldn't get a good place for it to look for an install).

Using the `BUILD_INTERFACE` and `INSTALL_INTERFACE` generators allow us a way of saying "when building, use this path, but when installing, use this other path." 

It was wrong of me to include it on `post-operation` -- the public header is coming from `XNNPACK`, so only `XNNPACK` needs to be configured with an understanding of where to find the header at build time, and where to put it during installation.